config/config.go

```diff
@@ -403,7 +376,7 @@ func (cfg *Manager) LoadJSON(bs []byte) error {
 			}
 			logger.Debugf("%s component configuration loaded", name)
 		} else {
-			cfg.undefinedComps[t.String()+name] = true
+			cfg.undefinedComps[t] = map[string]bool{name: true}
```

Comment:
This will re-assign a new map every time, overwriting any existing and thus losing information for other components in the same section.

I suggest you create an empty map everytime a section map is created (`RegisterComponent`).